From: Prashant Bhole <bhole_prashant_q7@lab.ntt.co.jp>
Date: Tue, 13 Feb 2018 13:44:22 +0900
Subject: bpf: samples/sockmap detach sock ops program
Patch-mainline: v4.17-rc1
Git-commit: 444890c3ce7d74bdc20f2bf930b8476b98d3e972
References: bsc#1109837

samples/sockops program keeps the sock_ops program attached to cgroup.
Fixed this by detaching program before exit.

Signed-off-by: Prashant Bhole <bhole_prashant_q7@lab.ntt.co.jp>
Signed-off-by: Alexei Starovoitov <ast@kernel.org>
Acked-by: Thomas Bogendoerfer <tbogendoerfer@suse.de>
---
 samples/sockmap/sockmap_user.c |    1 +
 1 file changed, 1 insertion(+)

--- a/samples/sockmap/sockmap_user.c
+++ b/samples/sockmap/sockmap_user.c
@@ -566,6 +566,7 @@ run:
 	else
 		fprintf(stderr, "unknown test\n");
 out:
+	bpf_prog_detach2(prog_fd[2], cg_fd, BPF_CGROUP_SOCK_OPS);
 	close(s1);
 	close(s2);
 	close(p1);
